contracts/WpxCvx.sol

```diff
@@ -0,0 +1,223 @@
+            CVX.safeApprove(oldCurvePool, 0);
+        }
+
+        // Set the approval on both wpxCVX and CVX for the new pool contract
+        allowance[address(this)][_curvePool] = type(uint256).max;
+        CVX.safeApprove(_curvePool, type(uint256).max);
+    }
+
+    /** 
+        @notice Set the reward receiver address
+        @param  _rewardReceiver  address  New reward receiver address
+     */
+    function setRewardReceiver(address _rewardReceiver) external onlyOwner {
+        if (_rewardReceiver == address(0)) revert ZeroAddress();
+
+        rewardReceiver = _rewardReceiver;
+
+        emit SetRewardReceiver(_rewardReceiver);
+    }
+
+    /**
+        @notice Redeem pxCVX snapshot rewards and transfer them to the currently set receiver
+        @param  epoch          uint256    Rewards epoch
+        @param  rewardIndexes  uint256[]  Reward indexes
+     */
+    function redeemRewards(uint256 epoch, uint256[] calldata rewardIndexes)
+        external
+    {
+        pirexCvx.redeemSnapshotRewards(epoch, rewardIndexes, rewardReceiver);
+    }
+
+    /** 
+        @notice Wrap the specified amount of pxCVX into wpxCVX
+        @param  amount  uint256  Amount of pxCVX
+     */
+    function wrap(uint256 amount) external nonReentrant {
+        if (amount == 0) revert ZeroAmount();
+
+        _mint(msg.sender, amount);
+
+        emit Wrap(msg.sender, amount);
+
+        pxCVX.safeTransferFrom(msg.sender, address(this), amount);
+    }
+
+    /** 
+        @notice Unwrap the specified amount of wpxCVX back into pxCVX
+        @param  amount  uint256  Amount of wpxCVX
+     */
+    function unwrap(uint256 amount) external nonReentrant {
+        if (amount == 0) revert ZeroAmount();
+
+        _burn(msg.sender, amount);
+
+        emit Unwrap(msg.sender, amount);
+
+        pxCVX.safeTransfer(msg.sender, amount);
+    }
+
+    /** 
+        @notice Swap the specified amount of source token into the counterpart token via the curvePool
+        @param  source       enum     Source token
+        @param  amount       uint256  Amount of source token
+        @param  minReceived  uint256  Minimum received amount of counterpart token
+     */
+    function swap(
+        Token source,
+        uint256 amount,
+        uint256 minReceived
+    ) external nonReentrant {
+        if (address(curvePool) == address(0)) revert PoolNotSet();
+        if (amount == 0) revert ZeroAmount();
+        if (minReceived == 0) revert ZeroAmount();
+
+        // We can infer both actual indices just by the cvxIndex value
+        uint256 index = cvxIndex;
+
+        if (source == Token.pxCVX) {
+            // Transfer the pxCVX to the contract and mint the equivalent amount of wpxCVX
+            pxCVX.safeTransferFrom(msg.sender, address(this), amount);
+            _mint(address(this), amount);
+
+            // Swap the wpxCVX for CVX and directly send to the user
+            uint256 received = curvePool.exchange(
```

Comment:
We can reduce lines of code by declaring `received` outside the conditionals' blocks, then assigning the result of `exchange` to it. The `Swap` emit can be consolidated to after the conditional logic is run.